src/System.Runtime.Extensions/ref/System.Runtime.Extensions.cs

```diff
@@ -8,14 +8,85 @@
 
 namespace System
 {
-   public partial class AppDomain
-   {
-       private AppDomain() {}
-       public static AppDomain CurrentDomain { get { return default(AppDomain); } }
-       public string BaseDirectory { get { return default(string); } }
-       public string RelativeSearchPath { get { return default(string); } }
-       public event System.UnhandledExceptionEventHandler UnhandledException { add { } remove { } } 
-   }
+    public partial class AppDomain : System.MarshalByRefObject
```

Comment:
How did you derive from MarshalByRefObject? It looks like it only has an internal constructor still in System.Runtime (https://github.com/dotnet/corefx/blob/master/src/System.Runtime/ref/System.Runtime.cs#L1421).